File: src/org/usfirst/frc5293/subsystems/ShooterKicker.java
package org.usfirst.frc5293.subsystems;

import edu.wpi.first.wpilibj.command.Subsystem;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
import org.usfirst.frc5293.Devices;
import org.usfirst.frc5293.commands.teleop.control.ShooterKickerControl;

public class ShooterKicker extends Subsystem {

    @Override
    protected void initDefaultCommand() {
        setDefaultCommand(new ShooterKickerControl());
    }

    public void set(double x) {
        SmartDashboard.putNumber("Shooter Kicker", x);
        Devices.getShooterKicker().getKicker().set(x);
    }

    public void setAngle(double angle) {
        SmartDashboard.putNumber("Shooter Kicker Angle", angle);
        Devices.getShooterKicker().getKicker().setAngle(angle);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/subsystems/ShooterKicker.kt
package org.usfirst.frc5293.subsystems

import edu.wpi.first.wpilibj.command.Subsystem
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
import org.usfirst.frc5293.Devices
import org.usfirst.frc5293.commands.teleop.control.ShooterKickerControl

class ShooterKicker : Subsystem() {

    private val servo by lazy { Devices.shooterKicker.kicker }

    override fun initDefaultCommand() {
        defaultCommand = ShooterKickerControl()
    }

    var value: Double
        get() = servo.get()
        set(x) {
            SmartDashboard.putNumber("Shooter Kicker", x)
            servo.set(x)
        }

    var angle: Double
        get() = servo.angle
        set(angle) {
            SmartDashboard.putNumber("Shooter Kicker Angle", angle)
            servo.angle = angle
        }
}